specs/language/introduction.tex

```diff
@@ -78,46 +128,148 @@
 architecture and the way they relate to the \acrshort{spmd} program model. In
 this document we will use the terms as defined in the following subsections.
 
-\Sub{\gls{lane}}{Intro.Model.Lane}
+\Sub{\acrshort{spmd} Terminology}{Intro.Model.Terms}
+
+\SubSub{Host and Device}{Intro.Model.Terms.HostDevice}
+
+\p \acrshort{hlsl} is a data-parallel programming language designed for
+programming auxiliary processors in a larger system. In this context the
+\textit{host} refers to the primary processing unit that runs the application
+which in turn uses a runtime to execute \acrshort{hlsl} programs on a supported
+\textit{device}. There is no strict requirement that the host and device be
+different physical hardware, although they commonly are. The separation of host
+and device in this specification is useful for defining the execution and memory
+model as well as specific semantics of language constructs.
+
+\SubSub{\gls{lane}}{Intro.Model.Terms.Lane}
 
 \p A \gls{lane} represents a single computed element in an \acrshort{spmd}
 program. In a traditional programming model it would be analogous to a thread of
 execution, however it differs in one key way. In multi-threaded programming
 threads advance independent of each other. In \acrshort{spmd} programs, a group
-of \gls{lane}s execute instructions in lock step because each instruction is a
-\acrshort{simd} instruction computing the results for multiple \gls{lane}s
-simultaneously.
+of \gls{lane}s may execute instructions in lockstep because each instruction may
+be a \acrshort{simd} instruction computing the results for multiple \gls{lane}s
+simultaneously, or synchronizing execution across multiple \gls{lane}s or
+\gls{wave}s. A \gls{lane} has an associated \textit{lane state} which denotes
+the execution status of the lane (\ref{Intro.Model.Terms.LaneState}).
+
+\SubSub{\gls{wave}}{Intro.Model.Terms.Wave}
 
-\Sub{\gls{wave}}{Intro.Model.Wave}
+\p A grouping of \gls{lane}s for execution is called a \gls{wave}. The size of a
```

Comment:
All lanes in a wave will execute in lockstep (right?) Wonder if that can be mentioned here.  Text above suggests that lanes are only running in lockstep when there's some kind of synchronization going on.

Reply:
They aren't required to be lock-step except at synchronization points. While it has been true with a lot of hardware that lock-step is guaranteed, that's less true with newer hardware. NV's latest GPUs have per-thread program counters which can allow much more creative divergence in execution.